tests/ident/test_engine_parser_comet_2020_01_04.py

```diff
@@ -106,69 +108,150 @@ def test_engine_parsers_comet_check_dataframe_integrity():
-                "358.174575",
-                None,
-                "comet_2020_01_4",
-                "2458",
-                None,
-                None,
-                "3",
-                "5.9000",
-                "1.0000",
-                "0.2825",
-                "3.76E+01",
-                "SHCIAEVEK;",
-                None,
-                "3",
-                "29",
-            ]
-        ]
-    ).all()
+
+def test_engine_parsers_comet_spec_records():
+    cv_param = etree.Element(
+        "cvParam",
+        cvRef="MS",
+        accession="MS:1001121",
+        name="number of matched peaks",
+        value="3",
+    )
+    spectrum_identification_item = etree.Element(
+        "SpectrumIdentificationItem",
+        id="SII_0",
+        rank="0",
+        chargeState="3",
+        peptide_ref="SHCIAEVEK;",
+        experimentalMassToCharge="358.174682",
+        calculatedMassToCharge="358.174575",
+        passThreshold="true",
+    )
+    spectrum_identification_result = etree.Element(
+        "SpectrumIdentificationResult",
+        id="SIR_16",
+        spectrumID="scan=2458",
+        spectraData_ref="SD0",
+    )
+    results = [cv_param, spectrum_identification_item, spectrum_identification_result]
+    obj = Comet_2020_01_4_Parser(input_file=None, params=None)
+
+    element_tag_prefix = "{http://psidev.info/psi/pi/mzIdentML/1.2}"
+
+    spec_results = {}
+    spec_ident_items = []
+    spec_records = []
+
+    for i in results:
+        entry = i
+        entry_tag = f"{element_tag_prefix}{i.tag}"
+        spec_results, spec_ident_items, spec_records = _spec_records(
+            entry,
+            entry_tag,
+            spec_results,
+            spec_ident_items,
+            spec_records,
+            obj.mapping_dict,
+        )
+    assert spec_records == [
+        {
+            "comet:num_matched_ions": "3",
+            "charge": "3",
+            "sequence": "SHCIAEVEK;",
+            "exp_mz": "358.174682",
+            "calc_mz": "358.174575",
+            "spectrum_id": "2458",
+        }
+    ]
+    assert len(spec_records[0]) == 6
+    assert spec_records[0]["comet:num_matched_ions"] == "3"
+    assert spec_records[0]["charge"] == "3"
+    assert spec_records[0]["sequence"] == "SHCIAEVEK;"
+    assert spec_records[0]["exp_mz"] == "358.174682"
+    assert spec_records[0]["calc_mz"] == "358.174575"
+    assert spec_records[0]["spectrum_id"] == "2458"
+
+
+def test_engine_parsers_comet_get_modifications():
+    cv_param = etree.Element(
+        "cvParam", cvRef="UNIMOD", accession="UNIMOD:UNIMOD:4", name="Carbamidomethyl"
+    )
+    search_modification = etree.Element(
+        "SearchModification", residues="C", massDelta="57.021464", fixedMod="true"
+    )
+    results = [cv_param, search_modification]
+    element_tag_prefix = "{http://psidev.info/psi/pi/mzIdentML/1.2}"
+
+    modification_mass_map = {}
+    mod_name = ""
+    fixed_mods = {}
+
+    for i in results:
```

Comment:
entry should be element and it should be` for element in (cv_param, search_modification)` imho